hokusai/lib/config.py

```diff
@@ -27,30 +27,62 @@ def check(self):
       raise HokusaiError("Hokusai is not set up for this project - run 'hokusai setup'")
     return self
 
-  def get(self, key):
-    self.check()
-    config_file = open(HOKUSAI_CONFIG_FILE, 'r')
-    config_data = config_file.read()
-    config_file.close()
-    config = yaml.safe_load(config_data)
+  def get(self, key, default=None, fallback_to_env=True, _type=str):
+    value = self.__value_for(key)
+    if value is not None:
+      if not isinstance(value, _type):
+        raise HokusaiError("Config key %s is not of %s" % (key, _type))
+      return value
+
+    if fallback_to_env is False:
+      return default
+
+    env_var_for_key = 'HOKUSAI_' + key.upper()
+    value_from_env = os.environ.get(env_var_for_key)
+    if value_from_env is None:
+      return default
+
     try:
-      return config[key]
-    except KeyError:
-      return None
+      return _type(value_from_env)
+    except ValueError:
+      raise HokusaiError("Environment variable %s could not be cast to %s" % (env_var_for_key, _type))
+
+
+  def __value_for(self, key):
+    self.check()
+    with open(HOKUSAI_CONFIG_FILE, 'r') as config_file:
+      config_struct = yaml.safe_load(config_file.read())
+      try:
+        return config_struct[key]
+      except KeyError:
+        pass
+      try:
+        return config_struct[key.replace('_', '-')]
```

Comment:
Well I effectively want to looking for either `foo_bar` or `foo-bar` in the config struct, with that precedence, but this order of precedence could be clearer... maybe a one-liner using `config_struct.keys()` to check for the available keys before accessing or `config_struct.get(k)` rather than `config_struct[k]` which returns `None` if no key, like Ruby, rather than raising a KeyError

Reply:
> maybe a one-liner using config_struct.keys() to check for the available keys before accessing or config_struct.get(k) rather than config_struct[k] which returns None if no key, like Ruby, rather than raising a KeyError

I do think this would be clearer.

Reply:
I dropped this as I felt like it was over-engineering to accept either underscores or hyphens for the same config keys (up to this point all config keys have been hyphenated anyway).  Added docs about this to `docs/Configuration Options.md`.